### PR TITLE
Masasa_fix_the_end_dates_issue

### PR DESCRIPTION
**Pull request by ThapeloMasasa**
# Description
Screenshot 2024-01-18 at 7 47 04 PM

## Related PRS (if any):
This PR is only frontend so it runs the back end on development branch

## Main changes explained:
- Added a check in case the person object has no end date and in which case the end date would be 'N/A'

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to dashboard→ Reports→ Reports→ People
6. verify that the end dates for each user is either a valid date that is not current date or is N/A

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73863365/69ea9668-2e0d-4dda-9767-0acb7fc5383e

### CODE REVIEW

File: package-lock.json
Comment (Nahiyan-16) [inline package-lock.json:1]: Noticed the deletion of package-lock.json. It's a crucial file for dependency management.

**Review by Nahiyan-16 — CHANGES_REQUESTED** [review]
## Hi there!
I believe you may have accidentally deleted the package-lock.json file which is crucial.
Assuming you didn't add any other dependencies, I tested your changes on the development package-lock.json file and your changes work accordingly.
Unavailable end dates show up as "N/A" while the correct end dates show for available ones.  

Screenshot 2024-01-23 104708

**Review by peterson337 — DISMISSED** [review]
The dates for users are either validated or marked as N/A.

a

**Review by 666saofeng — APPROVED** [review]
Hi, I tested with admin account and it worked as intended. Congrats!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62363348/8775b9be-62d8-43c4-83ae-f9f7247c4ed3

**Review by pshereen — DISMISSED** [review]
Tested the PR and the implemented feature is working as per the requirement.

Logged in as Admin user.
End dates for each user is either a valid date that is not current date or is N/A.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/7041610/c79b7157-b603-4d15-a229-65a3e37c69ac

**Review by SRamen1999 — DISMISSED** [review]
Tested the PR everything works as expected

Logged in as Admin user. End dates for each user is either a valid date that is not current date or is N/A. Everything is also updated accordingly when adding and deleting end dates.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56932664/20251a0c-d914-4d77-ac2c-f41f2dabc80f

**Review by Renanluizssx — DISMISSED** [review]
Hi @ThapeloMasasa, i tested your application and it's correct, nice

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/b4145fca-f2bd-4039-952e-120ff090fd8d

**Review by SaiDeepak-NEU — DISMISSED** [review]
I have logged in as an admin and checked the people tab under reports and the end dates are looking fine according to the description given. 
It is not showing the current date and showing the proper dates and displaying N/A for invalid fields.
Screenshot (27)
Screenshot (26)

**Review by Sophie-lei — DISMISSED** [review]
Hey, I just reviewed this PR.  I logged as an admin role and everything works as expected. The End Date has either the valid date or N/A. The User permission page also works well. Please check the video

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/70113822/34785624-c66c-4b80-bab8-ad66ca66ec16

A small problem would be looks like some end dates are easier than the start date. It may causes confusion. Is it better to limit the end date after the start date? like this 
Screenshot 2024-01-31 at 21 38 54

**Review by KevHnh — DISMISSED** [review]
I tested using an owner account and the changes are working as expected. The end dates are either "N/A" or displaying a valid date. 

image

**Comment by galgalinidhi** [comment]
Hi @ThapeloMasasa , I tested the PR and it works as expected! 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44816413/8555c2ea-c917-4ace-8a77-2662c2016b2a

**Review by ShengjieMao — DISMISSED** [review]
Hi, all dates seems to be fine and they have the same format.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56333697/f3ab178b-be89-408e-8170-af7ba982d5b0

**Comment by one-community** [comment]
Thank you all, moving to final review.

**Review by Renanluizssx — DISMISSED** [review]

**Comment by Aishwaryak01** [comment]
@ThapeloMasasa I can see that the checks have failed for this PR. Can you check?

**Pull request closed without merging**